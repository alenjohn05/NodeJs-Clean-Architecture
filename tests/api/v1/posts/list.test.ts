import { client } from '../..';

describe('GET /api/v1/posts', () => {
  describe('given an invalid request', () => {
    test('responds with a 400 status code', async () => {
      const pageSize = 1000; // Cannot exceed 50

      const response = await client.get(`/api/v1/posts?pageSize=${pageSize}`);

      expect(response.status).toEqual(400);
    });
  });

  describe('given a valid request', () => {
    test('responds with a 200 status code', async () => {
      const pageNumber = 3;
      const pageSize = 2;

      const response = await client.get(`/api/v1/posts?pageNumber=${pageNumber}&pageSize=${pageSize}`);

      expect(response.status).toEqual(200);
      expect(response.body).toMatchSnapshot();
    });
  });
});
