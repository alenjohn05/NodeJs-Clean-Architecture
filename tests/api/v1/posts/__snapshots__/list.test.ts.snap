// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /api/v1/posts given a valid request responds with a 200 status code 1`] = `
{
  "count": 6,
  "hasNextPage": false,
  "hasPreviousPage": true,
  "pageNumber": 3,
  "pageSize": 2,
  "posts": [
    {
      "id": 5,
      "published": true,
      "title": "Mock post five",
    },
    {
      "id": 6,
      "published": false,
      "title": "Mock post six",
    },
  ],
  "totalPages": 3,
}
`;
